### PR TITLE
Improve Bazel support

### PR DESCRIPTION
**Pull request by Vertexwahn**
This PR affects only the Bazel build support (CMake build stays untouched)

The following changes are introduced:

- Update Bazel version (7.1.1 -> 7.2.1)
- Update [platforms](https://github.com/bazelbuild/platforms) from 0.0.9 to 0.0.10
- Bazel tests: Bazilized a few more tests
- Documentation: Add Bzlmod variant

### CODE REVIEW

File: Bazel.md [hunk 1]
```diff
@@ -78,6 +80,16 @@ The expected output of this program is the current version of oneTBB.
 
 Switch to the folder with the files created earlier and run the binary with `bazel run //:Demo`.
 
+### Bzlmod
+
+If you are using Bzlmod you can fetch oneTBB using [the Bazel Central Registry](https://registry.bazel.build/).
```
Comment (aepanchi) [inline Bazel.md:85]:
```suggestion
If you use Bzlmod, you can fetch oneTBB with the [Bazel Central Registry](https://registry.bazel.build/).
```

File: Bazel.md [hunk 2]
```diff
@@ -78,6 +80,16 @@ The expected output of this program is the current version of oneTBB.
...
+
+Add to your `MODULE.bazel` file:
```
Comment (aepanchi) [inline Bazel.md:87]:
```suggestion
Add the following line to your `MODULE.bazel` file:
```

**Review by aepanchi — COMMENTED** [review]
Thanks! Just small suggestions :)

**Comment by isaevil** [comment]
@Vertexwahn,
Thank you for your regular contributions to oneTBB project towards Bazel support! Due to your standing contributions I would like to ask whether you be comfortable/have a desire to be put in CODEOWNERS files as an owner of Bazel-related code?
If so I would kindly ask you to update https://github.com/oneapi-src/oneTBB/blob/master/CODEOWNERS accordingly in the scope of this patch :)

**Comment by Vertexwahn** [comment]
@isaevil I added myself as a code owner for Bazel-related files

CI reports 

```
Unknown owner on line 17: make sure @Vertexwahn exists and has write access to the repository
/oneTBB/.bazelversion @Vertexwahn
```

I think I have no write permission to the repo... not sure if this is necessary for only giving reviews...

**Comment by isaevil** [comment]
> @isaevil I added myself as a code owner for Bazel-related files
> 
> CI reports
> 
> ```
> Unknown owner on line 17: make sure @Vertexwahn exists and has write access to the repository
> /oneTBB/.bazelversion @Vertexwahn
> ```
> 
> I think I have no write permission to the repo... not sure if this is necessary for only giving reviews...

Ok. We'll try to fix that in our next PRs.

**Review by isaevil — APPROVED** [review]
LGTM

**Comment by isaevil** [comment]
> @isaevil I added myself as a code owner for Bazel-related files
> 
> CI reports
> 
> ```
> Unknown owner on line 17: make sure @Vertexwahn exists and has write access to the repository
> /oneTBB/.bazelversion @Vertexwahn
> ```
> 
> I think I have no write permission to the repo... not sure if this is necessary for only giving reviews...

We'll figure this out :)

**Pull request merged**